sentry/src/main/java/io/sentry/SentryEvent.java

```diff
@@ -377,4 +368,9 @@ public boolean isCrashed() {
   public boolean isErrored() {
     return exception != null && !exception.getValues().isEmpty();
   }
+
+  @Override
+  public SentryItemType sentryItemType() {
```

Comment:
`instanceof SentryEvent` isn't the way to go?
I mean at the callsite, simply type check?

When serializing we'll need a field `type` to be either: `default`, `error` or `transaction`.
`error` is anyway inferred on the server if there's a stack trace (or an exception). but transaction will need to be set by the new class.